common/domain-setter.sub.html

```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>A page that will likely be same-origin-domain but not same-origin</title>
+
+<script>
+"use strict";
+document.domain = "{{host}}";
```

Comment:
In particular, it might be interesting to get the prototype object, then make it cross-origin by setting document.domain, and then see if you can still use that prototype object in the setter.